app/src/main/java/com/sopt/dive/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,375 @@
+    var idError by remember { mutableStateOf("") }
+    var pwError by remember { mutableStateOf("") }
+    var nicknameError by remember { mutableStateOf("") }
+    var drankError by remember { mutableStateOf("") }
+
+//    // Snackbar
+//    val scope = rememberCoroutineScope()
+//    val snackbarHostState = remember { SnackbarHostState() }
+
+
+    fun validateId(id: String): Boolean {
+        return when {
+            id.isEmpty() -> {
+                idError = "ID를 입력해주세요."
+                false
+            }
+            id.contains(" ") -> {
+                idError = "ID에는 공백을 사용할 수 없습니다."
+                false
+            }
+            id.length < 6 -> {
+                idError = "ID는 6글자 이상이어야 합니다."
+                false
+            }
+            id.length > 10 -> {
+                idError = "ID는 10글자 이하여야 합니다."
+                false
+            }
+            else -> {
+                idError = ""
+                true
+            }
+        }
+    }
+
+    fun validatePw(pw: String): Boolean {
+        return when {
+            pw.isEmpty() -> {
+                pwError = "비밀번호를 입력해주세요."
+                false
+            }
+            pw.contains(" ") -> {
+                pwError = "비밀번호에는 공백을 사용할 수 없습니다."
+                false
+            }
+            pw.length < 8 -> {
+                pwError = "PW는 8글자 이상이어야 합니다."
+                false
+            }
+            pw.length > 12 -> {
+                pwError = "PW는 12글자 이하여야 합니다."
+                false
+            }
+            else -> {
+                pwError = ""
+                true
+            }
+        }
+    }
+
+    fun validateNickname(nickname: String): Boolean {
+        return when {
+            nickname.isEmpty() -> {
+                nicknameError = "닉네임을 입력해주세요."
+                false
+            }
+            nickname.trim().isEmpty() -> {
+                nicknameError = "닉네임 공백은 사용할 수 없습니다."
+                false
+            }
+            nickname.isBlank() -> {
+                nicknameError = "닉네임은 한 글자 이상이어야 합니다."
+                false
+            }
+            else -> {
+                nicknameError = ""
+                true
+            }
+
+        }
+    }
+
+    fun validateDrink(drink: String): Boolean {
+        return when {
+
+            drink.isEmpty() -> {
+                drankError = "주량을 입력해주세요."
+                false
+            }
+            !drink.all { it.isDigit() } -> {
+                drankError = "숫자만 입력 가능합니다."
+                false
+            }
+            else -> {
+                drankError = ""
+                true
+            }
+
+        }
+    }
+
+    fun validateAll(): Boolean {
+        val isIdValid = validateId(idText)
+        val isPasswordValid = validatePw(pwText)
+        val isNicknameValid = validateNickname(nicknameText)
+        val isDrinkValid = validateDrink(drankText)
+
+        return isIdValid && isPasswordValid && isNicknameValid && isDrinkValid
+    }
```

Comment:
성공 조건을 함수로 잘 분리해놨으니, 저라면 함수 자체를 `validateId`가 아니라 `isValidId` 식으로 설정해두고 함수를 바로 호출해 리턴해줄 거 같아요.
```suggestion
    fun validateAll() = isValidId(idText) && isValidPassword(pwText) && isValidNickname(nicknameText) && isValidDrink(drankText)
```

Reply:
이렇게 하니까 휠씬 간결해지고 보기 좋은 것 같습니다! 감사합니다!